### PR TITLE
Feature/multiplebatteries

### PR DESCRIPTION
**Pull request by pkronstrom**

### CODE REVIEW

File: src/services/systemconfiguration.js [hunk 1]
```diff
@@ -46,40 +44,56 @@ class SystemConfiguration {
         return matches;
     }
 
+    captureOne(regex, str) {
```
Comment (pkronstrom) [inline src/services/systemconfiguration.js]: This is not used, so should be removed

File: src/nodes/input-battery.html [hunk 2]
```diff
@@ -21,37 +26,45 @@
...
+                    pathElem.val();
+            
+                    data.forEach(monitor => {
+                        serviceElem.append(
+                            $('<option/>', {value: JSON.stringify(monitor), text: monitor.name})
+                        );
+                    });
 
-                    let theBattery = data[Object.keys(data)[0]];
+                    let availablePaths = data[0]['paths'];
```
Comment (tkurki) [inline src/nodes/input-battery.html]: Add a comment / TODO here?

File: src/services/systemconfiguration.js [hunk 3]
```diff
@@ -46,40 +44,56 @@ class SystemConfiguration {
...
-
-            return services;
+        // parses all the dbus interfaces for batteries: com.victronenergy.battery.<id>
+        const re = /\bcom\.victronenergy\.battery\.(.*)/g;
+        
+        let dbusInterfaces = [...Object.keys(this.devices)];
+        let batteries = this.matchAndCapture(re, dbusInterfaces);
+        
+        let services = [];
+        batteries.forEach(battery => {
```
Comment (tkurki) [inline src/services/systemconfiguration.js]: `.map` would be clearer

File: src/services/victronclient.js [hunk 4]
```diff
@@ -10,8 +10,8 @@ const debug = require('debug')('node-red-contrib-victron:victronclient')
  * both subscribe to dbus value updates as well as write values to dbus.
  */
 class VictronClient {
-    constructor(dbusAddress) {
-        this.dbusAddress = dbusAddress || '127.0.0.1'
+    constructor(dbusAddress, dbusPort) {
+        this.dbusAddress = dbusAddress || '127.0.0.1:78'
```
Comment (tkurki) [inline src/services/victronclient.js]: Name shadow & confusion: is dbusAddress an ip address or addres + port?

File: src/nodes/config-client.js [hunk 5]
```diff
@@ -4,6 +4,12 @@ module.exports = function(RED) {
 
     var VictronClient = require('../services/victronclient.js');
 
+    function serialize(key, value) {
```
Comment (tkurki) [inline src/nodes/config-client.js]: maybe `setReplacer` (per JSON.stringify documentation, there it is called `replacer`), as it is not really serializing

Comment (tkurki) [reply src/nodes/config-client.js]: pure function, so move to top level scope?

File: src/services/victronclient.js (hunk 4 above)
Comment (pkronstrom) [reply src/services/victronclient.js]: Indeed. I think it's better to have separate dbus address and port, but have only one environment variable to set when running node-red. I'll use separate address and port in VictronClient, and split the port and address from the environment variable upon initializing the config node.

File: src/nodes/config-client.js (hunk 5 above)
Comment (pkronstrom) [reply src/nodes/config-client.js]: Good point, agreed.

File: src/nodes/input-battery.html (hunk 2 above)
Comment (pkronstrom) [reply src/nodes/input-battery.html]: The REST API (and the battery node as well) already supports for individual services per device. I will implement an onchange handler to change the measurement select to mirror the selected service.

**Pull request closed without merging**